server/routes/auth.js

```diff
@@ -0,0 +1,36 @@
+const express = require("express");
+const router = express.Router();
+
+const { signup, signin } = require("../validators/auth");
+const AuthService = require("../services/auth");
+const authService = new AuthService();
+
+// @route POST auth/signup
+// @desc User registration
+// @param name
+// @param email
+// @param password
+
+router.post("/signup", signup, async function (req, res, next) {
+  try {
+    const { newUser, token, cookieConfig } = await authService.signUpUser(
+      req.body
+    );
+    return res.status(201).cookie("token", token, cookieConfig).json(newUser);
+  } catch (error) {
+    next(error);
```

Comment:
Any specific reason to catch this error here vs let be picked up by the error handler?